SYCL/ESIMD/regression/copyto_char_test.cpp

```diff
@@ -0,0 +1,116 @@
+// REQUIRES: gpu
+// UNSUPPORTED: cuda || hip
+// RUN: %clangxx -fsycl %s -o %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+
+//==- copyto_char_test.cpp - Test for using copy_to to copy char buffers -==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <sycl/ext/intel/esimd.hpp>
+#include <sycl/sycl.hpp>
+
+#include <iostream>
+#include <memory>
+
+using namespace sycl::ext::intel::esimd;
+
+template <typename DataT>
+using shared_allocator = sycl::usm_allocator<DataT, sycl::usm::alloc::shared>;
+template <typename DataT>
+using shared_vector = std::vector<DataT, shared_allocator<DataT>>;
+
+using DataT = char;
+
+int main(int, char **) {
+  sycl::queue queue();
+
+  constexpr int NumElems = 10;

```

Comment:
Can you please outline the test code to a template routine with a parameter NumElems to call it with few more cases, i.e. with NumElems=10, (32+8), (32+10).
(32 here is the default ChunkSize in the implementation of copy_to() method)